lib/jobs/install-os.js

```diff
@@ -142,6 +117,30 @@ function installOsJobFactory(
     };
 
     /**
+     *
+     * @param {Object} options
+     * @param {Object} context
+     * @param {String} taskId
+     * @constructor
+     */
+    function InstallOsJob(options, context, taskId) {
+        var self = this;
+        InstallOsJob.super_.call(self, logger, options, context, taskId);
+
+        self.nodeId = self.context.target;
+        self.profile = self.options.profile;
+
+        //OS repository analyze job may pass some options via shared context
+        //The value from shared context will override the value in self options.
+        self.options = _.assign(self.options, context.repoOptions);
+
+        _validateOptions.call(self);
+        _convertOptions.call(self);
+        _encryptPassword.call(self);
+    }
+    util.inherits(InstallOsJob, BaseJob);
+
+    /**
```

Comment:
jshint yelled cause we are using functions that arent defined yet. So I moved the functions before we use them.